SwordPlugin.inc.php

```diff
@@ -35,15 +35,103 @@ public function register($category, $path, $mainContextId = null) {
 				HookRegistry::register('LoadComponentHandler', array($this, 'setupGridHandler'));
 				HookRegistry::register('EditorAction::recordDecision', array($this, 'callbackAuthorDeposits'));
+				// Preprints
+				HookRegistry::register('Publication::publish', array($this, 'callbackPublish'));
 			}
 			return true;
 		}
 		return false;
 	}
 
+	public function callbackPublish($hookName, $args) {
+		$newPublication =& $args[0];
+
+		if ($newPublication->getData('status') != STATUS_PUBLISHED) return false;
+		$submission = Services::get('submission')->get($newPublication->getData('submissionId'));
+
+		// Perform Automatic deposits
+		$request =& Registry::get('request');
+		$user = $request->getUser();
+		$context = $request->getContext();
+		$dispatcher = $request->getDispatcher();
+		$this->import('classes.PKPSwordDeposit');
+		$depositPointDao = DAORegistry::getDAO('DepositPointDAO');
+		$depositPoints = $depositPointDao->getByContextId($context->getId());
+		$sendDepositNotification = $this->getSetting($context->getId(), 'allowAuthorSpecify') ? true : false;
+		while ($depositPoint = $depositPoints->next()) {
+			$depositType = $depositPoint->getType();
+			if (($depositType == SWORD_DEPOSIT_TYPE_OPTIONAL_SELECTION)
+				|| $depositType == SWORD_DEPOSIT_TYPE_OPTIONAL_FIXED) {
+				$sendDepositNotification = true;
+			}
+			if ($depositType != SWORD_DEPOSIT_TYPE_AUTOMATIC)
+				continue;
+
+			try {
+				$deposit = new PKPSwordDeposit($submission);
+				$deposit->setMetadata($request);
+				$deposit->addEditorial();
+				$deposit->createPackage();
+				$deposit->deposit(
+					$depositPoint->getSwordUrl(),
+					$depositPoint->getSwordUsername(),
+					$depositPoint->getSwordPassword(),
+					$depositPoint->getSwordApikey()
+				);
+				$deposit->cleanup();
+			}
+			catch (Exception $e) {
+				$contents = __('plugins.importexport.sword.depositFailed') . ': ' . $e->getMessage();
+				$notificationMgr = new NotificationManager();
+				$notificationMgr->createTrivialNotification(
+					$user->getId(),
+					NOTIFICATION_TYPE_ERROR,
+					array('contents' => $contents)
+				);
+				error_log($e->getTraceAsString());
+			}
+
+			$user = $request->getUser();
+			$params = array(
+				'itemTitle' => $submission->getLocalizedTitle(),
+				'repositoryName' => $depositPoint->getLocalizedName()
+			);
+			$notificationMgr = new NotificationManager();
+			$notificationMgr->createTrivialNotification(
+				$user->getId(),
+				NOTIFICATION_TYPE_SUCCESS,
+				array('contents' => __('plugins.generic.sword.automaticDepositComplete', $params))
+			);
+		}
+
+		if ($sendDepositNotification) {
+			$submittingUser = $submission->getPrimaryAuthor();	// TODO how to get submission user ?
+			$contactName = $context->getSetting('contactName');
+			$contactEmail = $context->getSetting('contactEmail');
+
+			import('classes.mail.ArticleMailTemplate');
+			$mail = new ArticleMailTemplate($submission, 'SWORD_DEPOSIT_NOTIFICATION', null, $context, true);
```

Comment:
I think you should be able to use `SubmissionMailTemplate` directly here. ("Article" terminology in the preprint server code needs to be reviewed/fixed eventually.)